youtube_dl/extractor/letv.py

```diff
@@ -239,3 +240,75 @@ class LetvPlaylistIE(LetvTvIE):
         },
         'playlist_mincount': 7
     }]
+
+
+class LetvCloudIE(InfoExtractor):
+    IE_DESC = 'LetvCloud'
+    _VALID_URL = r'http://yuntv\.letv\.com/bcloud.html\?(?P<id>.*)$'
+
+    _TESTS = [{
+        'url': 'http://yuntv.letv.com/bcloud.html?uu=p7jnfw5hw9&vu=467623dedf',
+        'md5': '26450599afd64c513bc77030ad15db44',
+        'info_dict': {
+            'id': 'p7jnfw5hw9467623dedf',
+            'ext': 'mp4',
+            'title': 'p7jnfw5hw9467623dedf',
+            'timestamp': 1445922586,
+            'upload_date': '20151027',
+        },
+        'params': {
+            'format': '[width=1920]',
+        },
+    }, {
+        'url': 'http://yuntv.letv.com/bcloud.html?uu=p7jnfw5hw9&vu=ec93197892&pu=2c7cd40209&auto_play=1&gpcflag=1&width=640&height=360',
+        'info_dict': {
+            'id': 'p7jnfw5hw9ec93197892',
+            'ext': 'mp4',
+            'title': 'p7jnfw5hw9ec93197892',
+            'timestamp': 1450325068,
+            'upload_date': '20151217',
+        },
+    }, {
+        'url': 'http://yuntv.letv.com/bcloud.html?uu=p7jnfw5hw9&vu=187060b6fd',
+        'info_dict': {
+            'id': 'p7jnfw5hw9187060b6fd',
+            'ext': 'mp4',
+            'title': 'p7jnfw5hw9187060b6fd',
+            'timestamp': 1445946428,
+            'upload_date': '20151027',
+        },
+    }]
+
+    def _real_extract(self, url):
+        media_params = self._match_id(url)
+        uu = re.search("uu=([\w]+)", media_params).group(1)
+        vu = re.search("vu=([\w]+)", media_params).group(1)
+        media_id = uu + vu
+
+        play_json_req = sanitized_Request(
+            'http://api.letvcloud.com/gpc.php?cf=html5&sign=signxxxxx&ver=2.2&format=json&' +
+            "uu=" + uu + "&vu=" + vu)
+        play_json = self._download_json(play_json_req, media_id, 'Downloading playJson data')
+
+        urls = []
+        for v in play_json['data']['video_info']['media'].values():
+            url_info_dict = {
+                'url': base64.b64decode(v['play_url']['main_url']).decode("utf-8"),
+                'ext': 'mp4',
+                'format_id': int(v['play_url']['vtype']),
+                'format_note': v['play_url']['definition'],
+                'width': int(v['play_url']['vwidth']),
+                'height': int(v['play_url']['vheight']),
+            }
+            urls.append(url_info_dict)
+
+        publish_time = parse_iso8601(
+            play_json['data']['video_info']['add_time'],
+            delimiter=' ', timezone=datetime.timedelta(hours=8))
+
+        return {
+            'id': media_id,
+            'title': media_id,
+            'formats': urls,
```

Comment:
I guess format codes are some denotions not related to the quality. ```_sort_formats()``` will sort the formats by resolutions, bitrates, etc., so it would not be a problem. If formats are not sorted, users may download videos with worse qualities. For example in the case I've mentioned previously, 640x352 is downloaded instead the 1920x1072 one.